### PR TITLE
Fixes : cart widget + cart sidebar

### PR DESCRIPTION
**Pull request by patrixr**
Changes:

- Add `Show order summary` widget back into the order process
- Fix the cart item list within the widget
- Fix issue where clicking on `show` on the car sidebar right after requesting an item would close the sidebar instead of expanding it.

### CODE REVIEW

File: app/components/small-cart-toggle-view.js [hunk 1]
```diff
@@ -2,10 +2,21 @@ import Ember from "ember";
 
 export default Ember.Component.extend({
   showCartSummary: false,
+  allowClick: false,
+
+  application: Ember.computed(function() {
+    return Ember.getOwner(this).lookup("controller:application");
```
Comment (Darshan-Chauhan) [inline app/components/small-cart-toggle-view.js]: I don't think that it's a good idea to bind component with controller. 
You can inject a controller in a controller but ember specifically doesn't allow you to inject a controller in a component. 

Is there any other or better way to do this?

Comment (patrixr) [reply app/components/small-cart-toggle-view.js]: It's definitely not, but I'm trying to find the path of least resistance.
We have 4 or 5 different implementations of the cart, with just enough differences to make it hard to use a single component everywhere. The action is in the application, so my thought was either re-write it here, or call the one from the application.

I think both ways are equally bad here :/

Comment (patrixr) [reply app/components/small-cart-toggle-view.js]: any ideas ?

File: app/routes/application.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default Ember.Route.extend({
     if (error && _.isString(error)) {
       return error;
     }
-    return reason.errors[0].status == 403
+    return _.get(reason, "errors[0].status") == 403
```
Comment (abhinav098) [inline app/routes/application.js]: @patrixr why have we changed this ? I think the `reason` is passed to this function right ?

File: app/components/small-cart-toggle-view.js (hunk 1 above)
Comment (patrixr) [reply app/components/small-cart-toggle-view.js]: The main point is that I don't want to have yet another implementation of the list, which is why I'm using  `{{partial "cart_content/items"}}` inside the component

File: app/routes/application.js (hunk 2 above)
Comment (patrixr) [reply app/routes/application.js]: This was breaking when reason didn't have an `error` property.
Using `_.get` does the same thing excepts it won't crash if a nested property doesn't exist

Comment (patrixr) [reply app/routes/application.js]: It's the equivalent of doing `reason && reason.errors && reason.errors[0] && reasons.errors[0].status`

File: app/controllers/application.js [hunk 3]
```diff
@@ -40,7 +40,6 @@ export default Ember.Controller.extend(cancelOrderMixin, {
     return ["browse", "package_category"].indexOf(this.get("currentPath")) >= 0;
   }),
 
-  displayCart: false,
```
Comment (abhinav098) [inline app/controllers/application.js]: controllers/item.js has this action
```
    requestItem(item) {
      this.get("cart").add(item);
      Ember.run.later(
        this,
        function() {
          this.get("application").send("displayCart");
        },
        50
      );
    },
```

I hope removing this wont affect this action @patrixr

File: app/components/small-cart-toggle-view.js [hunk 4]
```diff
@@ -2,10 +2,21 @@ import Ember from "ember";
...
+  }),
 
   actions: {
     toggleCartSummary() {
       this.toggleProperty("showCartSummary");
+    },
+    showItemDetails(item) {
+      if (!this.get("allowClick")) {
```
Comment (Darshan-Chauhan) [inline app/components/small-cart-toggle-view.js]: Ammm what is the use of `allowClick` here?
The initial value if defaulting to false and I don't see it changing anywhere else in this component.
So `!this.get("allowClick")` will be always true

Comment (Darshan-Chauhan) [reply app/components/small-cart-toggle-view.js]: Also that property is not passed to component either *{{{{small-cart-toggle-view}}}}*

Comment (patrixr) [reply app/components/small-cart-toggle-view.js]: It's for the ability to turn it on later on. We had a long discussion with Matt on whether we wanted to allow user to navigate away from the order process in the middle (by clicking on an item)

Comment (patrixr) [reply app/components/small-cart-toggle-view.js]: We can remove it if you don't like having an un-used feature, I don't mind

File: app/components/small-cart-toggle-view.js (hunk 1 above)
Comment (Darshan-Chauhan) [reply app/components/small-cart-toggle-view.js]: Will it be possible to move ** showItemDetails** action from application controller to `cart-service` and inject that service in application controller(might be there already) and component?

File: app/components/small-cart-toggle-view.js (hunk 4 above)
Comment (Darshan-Chauhan) [reply app/components/small-cart-toggle-view.js]: I wasn't aware about the discussion with Matt 😄 
I don't mind keeping un-used features but it just hard to understand at first glance.
Can you add a similar comment so it'll be easier to understand for everyone?

**Comment by patrixr** [comment]
@Darshan-Chauhan Updated 👍

**Pull request merged**